proto/redpanda/api/dataplane/v1alpha1/kafka_connect.proto

```diff
@@ -0,0 +1,328 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/empty.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+// The following states are possible for a connector or one of its tasks
+// implement the state interface described in the Kafka connect API @see
+// https://docs.confluent.io/platform/current/connect/monitoring.html#connector-and-task-status
+// this includes holistic unified connector status that takes into account not
+// just the connector instance state, but also state of all the tasks within the
+// connector
+enum HolisticState {
```

Comment:
Maybe `ConnectorHolisticState` or just `ConnectorState`. The name HolisticState for the entire API package may be too generic.

Reply:
ConnectorState may be clashing with the state that Kafka connect already reports for a Connector though. This state is kind of the reason why we introduced our "own state" that considers the task states as well as the connector states.